astropy/modeling/core.py

```diff
@@ -1546,7 +1846,7 @@ def _check_param_broadcast(self, params, max_ndim):
                 "to the broadcasting rules.".format(param_a, shape_a,
                                                     param_b, shape_b))
 
-    def _param_sets(self, raw=False):
+    def _param_sets(self, raw=False, units=False):
```

Comment:
I think the shape is still equivalent, it's just that you have to nest the 1D quantities inside a 1D array rather than have a 2D array. We can't return 2D quantities with mismatching units.